### PR TITLE
Obs matrix work

### PR DESCRIPTION
**Pull request by keskitalo**
Support for writing out noise-weighted observation matrices for rapid co-adding.

### CODE REVIEW

**Comment by keskitalo** [comment]
Unit test added and passes.

File: src/toast/tests/ops_filterbin.py [hunk 1]
```diff
@@ -616,3 +617,156 @@ def test_filterbin_obsmatrix_cached(self):
...
+            close_data(obs_data)
+
+        if data.comm.world_rank == 0:
+            import matplotlib.pyplot as plt
+
+            # Assemble the single-run matrix
+
+            rootname = os.path.join(self.outdir, f"split_run_obs_matrix")
+            #try:
+            fname_matrix = ops.combine_observation_matrix(rootname)
```
Comment (tskisner) [inline src/toast/tests/ops_filterbin.py:736]: I don't see where this is defined in the `toast.ops` module.  Maybe a typo or a file is not committed?

**Review by tskisner — CHANGES_REQUESTED** [review]
Looks good, just need to add the missing entry point and check out whether there is a typo or missing file (or tell me what I missed).

File: src/toast/_libtoast/todmap_mapmaker.cpp [hunk 2]
```diff
@@ -88,21 +88,142 @@ void project_signal_offsets(py::array_t <double> ref, py::list todslices,
     }
 }
 
+void add_matrix(
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/_libtoast/todmap_mapmaker.cpp:91]: Changed the name

File: src/toast/scripts/CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@ set(ENTRY_POINTS
     toast_hdf5_to_spt3g.py
     toast_mini.py
     toast_obsmatrix_combine.py
+    toast_obsmatrix_coadd.py
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/scripts/CMakeLists.txt:18]: Added

File: src/toast/tests/ops_filterbin.py [hunk 4]
```diff
@@ -616,3 +617,156 @@ def test_filterbin_obsmatrix_cached(self):
...
+            fnames = glob.glob(
+                f"{self.outdir}/{orig_name_filterbin}*noiseweighted_obs_matrix*"
+            )
+            rootnames = set()
+            for fname in fnames:
+                rootnames.add(fname.split(".")[0])
+
+            filenames = []
+            for rootname in rootnames:
+                fname_matrix = ops.combine_observation_matrix(rootname)
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/tests/ops_filterbin.py:752]: see above

File: src/toast/tests/ops_filterbin.py (hunk 1 above)
Comment (tskisner) [reply src/toast/tests/ops_filterbin.py:736]: Sorry, completely missed that import in `toast.ops.__init__.py`!

**Comment by keskitalo** [comment]
Accidentally applied the corrections to the `toast3` branch. Rebased this to match.

**Review by tskisner — APPROVED** [review]
Thanks for the fixes!

**Comment by keskitalo** [comment]
Merging the approved PR now that all of the CI tests pass.

**Pull request merged**